lib/Service/FileService.php

```diff
@@ -575,7 +575,8 @@ public function createShareLink(string $path, ?int $shareType = 3, ?int $permiss
 
 		try {
 			// Note: if we ever want to create share links for folders instead of files, just remove this try catch and only use setTarget, not setNodeId.
-			$file = $userFolder->get(path: $path);
+            $file = $this->rootFolder->get($path);
```

Comment:
Could this not break on the above code if we for example don't have a user? 🤔 

![image](https://github.com/user-attachments/assets/9173cb22-e44e-4a70-911f-b289335b067e)

Reply:
In the new structure, we always have a user ;).